src/compas/datastructures/assembly/part.py

```diff
@@ -2,23 +2,92 @@
 from __future__ import absolute_import
 from __future__ import division
 
-from collections import deque
+from compas.geometry import Frame
+from compas.geometry import Brep
+from compas.geometry import Polyhedron
+from compas.datastructures import Datastructure
+from compas.data import Data
 
-# from functools import reduce
 
-from compas.geometry import Frame
-from compas.geometry import Polyhedron as Shape
-from compas.geometry import Transformation
+class Feature(Data):
+    """Base class for a feature which may be applied to a :class:`~compas.datastructures.Part`."""
+
+    def apply(self, part):
+        """Apply this Feature to the given part.
+
+        Parameters
+        ----------
+        part : :class:`~compas.datastructures.Part`
+            The part onto which this feature should be applied.
+
+        """
+        raise NotImplementedError
+
+
+class GeometricFeature(Feature):
+    """Base class for geometric feature which may be applied to a :class:`~compas.datastructures.Part`.
+
+    Applies a binary operation on Part's current geometry and the feature's.
+
+    An implementation of this class may offer support for various geometry types by adding an entry to OPERATIONS
+    mapping the geometry type to its corresponding operation.
+
```

Comment:
Perhaps it would be good to add an example, so implementers know what is expected/intended...